packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,354 @@
+   * Columns will be presented in the order given.
+   */
+  pinnedColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+   * Data for overflow columns. Overflow columns are rendered in the table's horizontal overflow.
+   * Columns will be presented in the order given.
+   */
+  overflowColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+   * Number indicating the default column width in px. This value will be used if no overriding width value is provided on a per-column basis.
+   */
+  defaultColumnWidth: PropTypes.number,
+
+  /**
+   * String that specifies the column height. Any valid CSS height value is accepted.
+   */
+  columnHeaderHeight: PropTypes.string,
+
+  /**
+   * String that specifies the height for the rows in the table. Any valid CSS value is accepted.
+   */
+  rowHeight: PropTypes.string,
+
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the table.
+   */
+  rowHeaderIndex: validateRowHeaderIndex,
+
+  /**
+   * Function that is called when a resizable column is resized. Parameters:
+   * @param {string} columnId columnId
+   * @param {string} requestedWidth requestedWidth
+   */
+  onColumnResize: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable cell is selected. Parameters:
+   * @param {string} rowId rowId
+   * @param {string} columnId columnId
+   */
+  onCellSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable column is selected. Parameters:
+   *  @param {string} columnId columnId
+   */
+  onColumnSelect: PropTypes.func,
+
+  /**
+   * Boolean indicating whether or not the DataGrid should allow entire rows to be selectable. An additional column will be
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  defaultColumnWidth: 200,
+  columnHeaderHeight: '2.5rem',
+  rowHeight: '2.5rem',
+  pinnedColumns: [],
+  overflowColumns: [],
+  rows: [],
+};
+
+const defaultColumnMinimumWidth = 60;
+const defaultColumnMaximumWidth = 300;
+
+function Table(props) {
+  const {
+    id,
+    ariaLabelledBy,
+    ariaLabel,
+    rows,
+    pinnedColumns,
+    overflowColumns,
+    onColumnResize,
+    defaultColumnWidth,
+    columnHeaderHeight,
+    rowHeight,
+    onColumnSelect,
+    onCellSelect,
+    hasSelectableRows,
+    rowHeaderIndex,
+    intl,
+  } = props;
+
+  if (pinnedColumns.length === 0) {
+    // eslint-disable-next-line no-console
+    console.warn(ERRORS.PINNED_COLUMNS_UNDEFINED);
+  }
+
+  // Manage column resize
+  const [tableHeight, setTableHeight] = useState(0);
+  const [activeIndex, setActiveIndex] = useState(null);
+  const [cellAriaLiveMessage, setCellAriaLiveMessage] = useState(null);
+
+  const [pinnedColumnOffsets, setPinnedColumnOffsets] = useState([0]);
```

Comment:
At some point I'd love to re-factor this and move the code around but like the table columns logic this is non-trivial to change since AriaLiveMessage is used pretty much everywhere for the screen reader. We could probably log another JIRA to do this or flesh out the re-factor in another story.

Reply:
Makes sense. That's fine then.